tests/Doctrine/Tests/ORM/Functional/Ticket/DDC3609Test.php

```diff
@@ -0,0 +1,286 @@
+
+          LEFT JOIN ComplexOrder.deliveryLocations ComplexDeliveryLocation
+
+        WHERE
+          (
+            BaseOrder INSTANCE OF Doctrine\Tests\ORM\Functional\Ticket\DDC3609SimpleOrder
+            AND
+            SimpleOrder.fulfiller = 'FulfillerUser'
+          )
+          OR
+          (
+            BaseOrder INSTANCE OF Doctrine\Tests\ORM\Functional\Ticket\DDC3609ComplexOrder
+            AND
+            ComplexDeliveryLocation.fulfiller = 'FulfillerUser'
+          )
+        ";
+
+        $query = $this->_em->createQuery($dql);
+        $results = $query->execute();
+
+        $foundSimple = false;
+        $foundComplex = false;
+        foreach ($results as $result) {
+            if ($result->id == $shouldNotAppear->id) {
+                $this->fail('Order was present in results when it should not have been');
+            }
+            if ($result->id == $simpleOrder->id) {
+                $foundSimple = true;
+            }
+            if ($result->id == $complexOrder->id) {
+                $foundComplex = true;
+            }
+        }
+
+        $this->assertTrue($foundSimple, 'Simple order was not found in search results');
+        $this->assertTrue($foundComplex, 'Complex order was not found in search results');
+    }
+
+    /**
+     * Verifies that a SimpleOrder can be found by selecting the base class and
+     * only querying on fields that are on the base class.
+     */
+    public function testCanFindSimpleOrderByBaseField()
+    {
+        $orderToFind = new DDC3609SimpleOrder();
+        $orderToFind->requester = 'RequesterUser';
+
+        $this->_em->persist($orderToFind);
+        $this->_em->flush();
+
+        $dql = "
+        SELECT
+          BaseOrder
+
+        FROM
+          Doctrine\Tests\ORM\Functional\Ticket\DDC3609Order BaseOrder
+
+        WHERE
+          BaseOrder.requester = 'RequesterUser'
+        ";
+
+        $query = $this->_em->createQuery($dql);
+        $results = $query->execute();
+
+        $foundOrder = false;
+        foreach ($results as $result) {
+            if ($result->id == $orderToFind->id) {
+                $foundOrder = true;
+            }
+        }
+
+        $this->assertTrue($foundOrder, 'Order was not found in search results');
+    }
+
+    /**
+     * Tests that a property on the SimpleOrder (one level of inheritance) can be
+     * queried on.
+     */
+    public function testCanFindSimpleOrderByChildField()
+    {
+        $orderToFind = new DDC3609SimpleOrder();
+        $orderToFind->fulfiller = 'FulfillerUser';
+
+        $this->_em->persist($orderToFind);
+        $this->_em->flush();
+
+        $dql = "
+        SELECT
+          BaseOrder
+
+        FROM
+          Doctrine\Tests\ORM\Functional\Ticket\DDC3609Order BaseOrder
+          LEFT JOIN Doctrine\Tests\ORM\Functional\Ticket\DDC3609SimpleOrder SimpleOrder WITH SimpleOrder.id = BaseOrder.id
```

Comment:
You're right for normal use, however `testCanFindSimpleOrderByChildField` and `testCanFindComplexOrder` were added to break down the failing query a bit more to help with debugging. The query in `testIssue` is the real use case.

To be clear, my goal is to get a result set of `SimpleOrder`s and  `ComplexOrder`s where the fulfiller is 'FulfillerUser`. Since these objects both have the concept of a fulfiller (even though they store it via different relationships) I'd like to do this in a single query.